app/controllers/close_contacts_controller.rb

```diff
@@ -23,35 +19,95 @@ def create
-
-    cc = CloseContact.find_by(id: params.permit(:id)[:id])
-    cc.update(first_name: params.permit(:first_name)[:first_name],
-              last_name: params.permit(:last_name)[:last_name],
-              primary_telephone: params.permit(:primary_telephone)[:primary_telephone],
-              email: params.permit(:email)[:email],
-              last_date_of_exposure: params.permit(:last_date_of_exposure)[:last_date_of_exposure],
-              assigned_user: params.permit(:assigned_user)[:assigned_user],
-              notes: params.permit(:notes)[:notes],
-              contact_attempts: params.permit(:contact_attempts)[:contact_attempts])
-    cc.save
-    History.close_contact_edit(patient: patient_id,
-                               created_by: current_user.email,
-                               comment: "User edited a close contact (ID: #{cc.id}).")
+    update_params = {
+      first_name: params.permit(:first_name)[:first_name],
+      last_name: params.permit(:last_name)[:last_name],
+      primary_telephone: params.permit(:primary_telephone)[:primary_telephone],
+      email: params.permit(:email)[:email],
+      last_date_of_exposure: params.permit(:last_date_of_exposure)[:last_date_of_exposure],
+      assigned_user: params.permit(:assigned_user)[:assigned_user],
+      notes: params.permit(:notes)[:notes],
+      contact_attempts: params.permit(:contact_attempts)[:contact_attempts]
+    }
+    ActiveRecord::Base.transaction do
+      if @close_contact.update(update_params)
+        History.close_contact_edit(patient: @patient.id,
+                                   created_by: current_user.email,
+                                   comment: "User edited a close contact (ID: #{@close_contact.id}).")
+      else
+        # Handle case where close contact update failed
+        error_message = 'Close Contact was unable to be updated.'
+        render(json: { error: error_message }, status: :bad_request) && return
+      end
+    end
+  end
+
+  # Delete an existing close contact record
+  def destroy
+    ActiveRecord::Base.transaction do
+      if @close_contact.destroy
+        reason = params.require(:delete_reason)
+        comment = "User deleted a close contact (ID: #{@close_contact.id}"
+        comment += ", Name: #{@close_contact.first_name} #{@close_contact.last_name}" unless (@close_contact.first_name + @close_contact.last_name).blank?
+        comment += ", Primary Telephone: #{@close_contact.primary_telephone}" unless @close_contact.primary_telephone.blank?
+        comment += ", Email: #{@close_contact.email}" unless @close_contact.email.blank?
+        comment += ", Last Date of Exposure: #{@close_contact.last_date_of_exposure}" unless @close_contact.last_date_of_exposure.blank?
+        comment += ", Assigned User: #{@close_contact.assigned_user}" unless @close_contact.assigned_user.blank?
+        comment += ", Notes: #{@close_contact.notes}" unless @close_contact.notes.blank?
```

Comment:
this might be a question for the PO team, but including this makes the history fairly long.  This is not included for the vaccines delete, so I think either way it should be the same for both.  I'll post in the ui channel

Reply:
you're good on this, I need to add notes to mine